transport/src/main/java/io/netty/channel/local/LocalChannel.java

```diff
@@ -296,6 +297,29 @@ private void readInbound() {
             if (received == null) {
                 break;
             }
+            if (received instanceof ByteBuf) {
+                ByteBuf msg = (ByteBuf) received;
+                ByteBuf output = handle.allocate(alloc());
```

Comment:
I wonder if we should maybe only allocate and do the memory copy if `inboundBuffer.peak() instanceof ByteBuf` returns true. WDYT ?

Reply:
Sounds like a good idea for both branches.